src/xrpld/app/tx/detail/Transactor.cpp

```diff
@@ -180,6 +188,51 @@ Transactor::calculateBaseFee(ReadView const& view, STTx const& tx)
     return baseFee + (signerCount * baseFee);
 }
 
+TER
+Transactor::checkPermissions(
+    ReadView const& view,
+    STTx const& tx,
+    std::unordered_set<GranularPermissionType>& permissions)
+{
+    if (!view.read(keylet::account(tx[sfOnBehalfOf])))
```

Comment:
Should be `exists` instead.

Should we check if `sfOnBehalfOf` is included in `tx`?  `checkPermissions` is so far only called in delegated but still might make sense to check.